numba/misc/dummyarray.py

```diff
@@ -314,6 +314,27 @@ def reshape(self, *newdims, **kws):
         if order not in 'CFA':
             raise ValueError('order not C|F|A')
 
+        # check for exactly one instance of -1 in newdims
```

Comment:
```suggestion
        # check for exactly one instance of -1 in newdims
        # https://github.com/numpy/numpy/blob/623bc1fae1d47df24e7f1e29321d0c0ba2771ce0/numpy/core/src/multiarray/shape.c#L470-L515   # noqa: E501
```

Please could you add the above as this is a translation from NumPy? Thanks.